migrations/000032_app_versions_table.up.sql

```diff
@@ -0,0 +1,489 @@
+BEGIN;
+
+SET search_path = public, pg_catalog;
+
+--
+-- Drop obsolete views that will not work with versions.
+--
+DROP VIEW IF EXISTS app_job_types;
+DROP VIEW IF EXISTS vice_analyses;
+
+--
+-- Drop views that depend on columns which will be dropped,
+-- or that will add new view columns.
+--
+DROP VIEW IF EXISTS tool_listing;
+DROP VIEW IF EXISTS app_listing;
+DROP VIEW IF EXISTS job_listings;
+
+--
+-- app_versions table
+--
+CREATE TABLE IF NOT EXISTS app_versions (
+    id uuid NOT NULL DEFAULT uuid_generate_v1(),
+    app_id uuid NOT NULL,
+    version varchar(255) NOT NULL,
+    version_order int NOT NULL DEFAULT 0,
+    deleted boolean NOT NULL DEFAULT false,
+    disabled boolean NOT NULL DEFAULT false,
+    integration_data_id uuid NOT NULL,
+    integration_date timestamp without time zone,
+    edited_date timestamp without time zone DEFAULT CURRENT_TIMESTAMP,
+
+    PRIMARY KEY (id),
+    FOREIGN KEY (app_id) REFERENCES apps(id) ON DELETE CASCADE,
+    UNIQUE (app_id, version_order)
+);
+
+CREATE INDEX IF NOT EXISTS app_versions_app_id ON app_versions(app_id);
+CREATE INDEX IF NOT EXISTS app_versions_version_order ON app_versions(version_order);
```

Comment:
We should look at the actual queries we're doing (which is to say, I should, you already have, lol), but I'm wondering if this index will be useful separately like this. I'm guessing most queries will either limit or join on the `app_id` column, so they'd probably use the index created by the `UNIQUE`

I also wonder if we should have an index on `(app_id, version)` (which would happen with the UNIQUE I mentioned, but we might not add that) and/or app_id plus version and version_order, since that might be useful for some queries listing versions (but it depends on the queries even more for that one)

Reply:
Right, good point. I don't think we actually need the separate index on the `version_order`.

I don't think we need an index that includes the `version` either, since I envisioned the API endpoints to look like `/apps/:app-id/versions/:version-id`, so I don't think we will need to query on the version labels. They should only need to be displayed when fetching info on a specific version (looked up by version ID), or in a complete listing of an app's versions, sorted by `version_order`.

Reply:
Yeah -- for the second index I mentioned, the place where it could be useful is if we have something that will fetch the app ID, version id, and version string, but nothing else from the table, so the index could be used in an index-only scan. Looking at it again I think that's fairly unlikely.